### ISSUE TITLE
c3c9d6b15f breaks build with clang++ as C-compiler

### ISSUE THREAD

**Issue by jkeenan** [Perl/perl5#19515]
The following commit has caused `make` to fail when compiling `perl` using `clang++` as the C-compiler:
```
commit c3c9d6b15f57fdce79988a553671a1ceb54c0f10
Author:     Yves Orton <demerphq@gmail.com>
AuthorDate: Sun Mar 6 09:42:14 2022 +0100
Commit:     Yves Orton <demerphq@gmail.com>
CommitDate: Thu Mar 10 03:22:33 2022 +0100

    hv.c - rework PL_hash_rand_bits update logic, add tests, -Dh debug mode
```
This was first observed in smoke-test reports filed by Carlos Rivera here:

[FreeBSD-12/clang++-10.0.1](https://perl5.test-smoke.org/report/5016686)

[FreeBSD-14/clang++-13.0.0](https://perl5.test-smoke.org/report/5016693)

I confirmed this using clang++-10.0.1 on FreeBSD-12.  At the commit immediately preceding the one above, 8a35994ac8, `make` completes successfully (albeit with lots of build-time warnings and test failures -- but those are better discussed separately).  At c3c9d6b15f, however, `make` quickly fails with output like this:
```
clang++ -c -DPERL_CORE -DHAS_FPSETMASK -DHAS_FLOATINGPOINT_H -I/usr/local/include -O2 -Wall -Werror=pointer-arith -Werror=vla -Wextra -Wno-long-long -Wno-declaration-after-statement -Wc++-compat -Wwrite-strings util.c
clang++: warning: treating 'c' input as 'c++' when in C++ mode, this behavior is deprecated [-Wdeprecated]
util.c:5079:41: error: invalid suffix on literal; C++11 requires a space between literal and identifier [-Wreserved-user-defined-literal]
                        " RAND_BITS=0x%"UVxf, PL_hash_rand_bits);
                                        ^

1 error generated.
*** Error code 1

```
(Complete output available upon request.)

Here is the relevant part of `util.c`:
```
5055 void
5056 Perl_debug_hash_seed(pTHX_ bool via_debug_h)
5057 {
...
5071 #ifdef PERL_HASH_RANDOMIZE_KEYS
5072             PerlIO_printf(Perl_debug_log, " PERTURB_KEYS = %d (%s)",
5073                     PL_HASH_RAND_BITS_ENABLED,
5074                     PL_HASH_RAND_BITS_ENABLED == 0 ? "NO" :
5075                     PL_HASH_RAND_BITS_ENABLED == 1 ? "RANDOM"
5076                                                    : "DETERMINISTIC");
5077             if (DEBUG_h_TEST)
5078                 PerlIO_printf(Perl_debug_log,
5079                         " RAND_BITS=0x%"UVxf, PL_hash_rand_bits);
5080 #endif
5081             PerlIO_printf(Perl_debug_log, "\n");
5082         }
5083     }
5084 #endif /* #if (defined(USE_HASH_SEED) ... */
5085 }
```
@demerphq, can you take a look?

**Comment by Leont** [comment]
The solution is trivial: just add a space between the string and the `UVxf`

**Comment by jkeenan** [comment]
This should be fixed by https://github.com/Perl/perl5/pull/19516.  I'll monitor for a day or two before closing.

**Comment by jkeenan** [comment]
> This should be fixed by #19516. I'll monitor for a day or two before closing.

Smoke reports look good; closing ticket.
